pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -450,17 +492,59 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListenerForPortOrUDS(no
 		// add one empty entry to the list so we generate a default listener below
 		allChains = []plugin.FilterChain{{}}
 	}
-	for _, chain := range allChains {
+
+	// Detect protocol by sniffing and double the filter chain
+	if pluginParams.ListenerProtocol == plugin.ListenerProtocolAuto {
```

Comment:
What if we are a 1.2 proxy? is it impossible to have protocol auto at this point?

Reply:
If user is using 1.2 proxy, they have no way to set listener protocol to `Auto`. `Auto` is only used when proxy >= 1.3